src/firefly/js/visualize/draw/CanvasWrapper.jsx

```diff
@@ -88,9 +90,40 @@ class CanvasWrapper extends React.Component {
     constructor(props) {
         super(props);
         this.drawer= null;
+        this.lastDrawLayer= null;
+        
     }
 
-    shouldComponentUpdate(np,ns) { return sCompare(this,np,ns); }
+    // shouldComponentUpdate(np,ns) {
+    //     return sCompare(this,np,ns);
+    // }
+
+
+    shouldComponentUpdate(nProps) {
+
+        var {plot,drawLayer,width,height}= nProps;
+        const p= this.props;
+
+        const update= (width!==p.width || height!==p.height || plot!==p.plot);
+
+        if (!update) {
+            this.updateDrawLayer(nProps);
+        }
```

Comment:
I may not understand the logic, 
 this.updateDrawLaywer is done when 'update' is 'false'?